modules/apps/29-fee/ibc_middleware.go

```diff
@@ -267,12 +267,25 @@ func (im IBCMiddleware) OnAcknowledgementPacket(
 
 	packetID := channeltypes.NewPacketId(packet.SourcePort, packet.SourceChannel, packet.Sequence)
 	feesInEscrow, found := im.keeper.GetFeesInEscrow(ctx, packetID)
-	if found {
+	if !found {
+		// call underlying callback
+		return im.app.OnAcknowledgementPacket(ctx, packet, ack.AppAcknowledgement, relayer)
+	}
+
+	payee, found := im.keeper.GetPayeeAddress(ctx, relayer.String(), packet.SourceChannel)
+	if !found {
 		im.keeper.DistributePacketFeesOnAcknowledgement(ctx, ack.ForwardRelayerAddress, relayer, feesInEscrow.PacketFees, packetID)
+		return nil
+	}
+
+	payeeAddr, err := sdk.AccAddressFromBech32(payee)
+	if err != nil {
+		return sdkerrors.Wrapf(err, "failed to create sdk.Address from payee: %s", payee)
 	}
 
-	// call underlying callback
-	return im.app.OnAcknowledgementPacket(ctx, packet, ack.AppAcknowledgement, relayer)
+	im.keeper.DistributePacketFeesOnAcknowledgement(ctx, payeeAddr.String(), payeeAddr, feesInEscrow.PacketFees, packetID)
```

Comment:
Nice catch! I'll push an update with the option to distribution to the `ack.ForwardRelayerAddress` as we discussed.
It will reduced the number of store lookups and txs required in total that way.